runbot_gitlab_minimal/controllers/gitlab_ci_controller.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Copyright <2017> <Vauxoo info@vauxoo.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from openerp import http, SUPERUSER_ID
+from openerp.http import request
+from openerp.addons.runbot import runbot
+
+_logger = logging.getLogger(__name__)
+
+
+class RunbotCIController(runbot.RunbotController):
+
+    @http.route(['/runbot/hook_gitlab/<int:repo_id>',
+                 '/runbot/hook_gitlab/org'], type='json', auth="public",
+                website=True)
+    def hook_gitlab(self, repo_id=None, **post):
+        data = request.jsonrequest
+        event = data['object_kind'] if data.has_key('object_kind') else None
+        repository = data['repository']
+        if repo_id is None:
+            if event in ['push', 'merge_request']:
```

Comment:
You have duplicated code here from [original runbot hook](https://github.com/Vauxoo/odoo-extra/blob/397ba6d17e474f9b841860fcf0aae8a8243d8903/runbot/runbot.py#L1311-L1313)

You could use a mapped dictionary keys github 2 gitlab here too